### PR TITLE
DApp3 User Profile/Identity page

### LINKED ISSUE #2415: Improved Profile/Identity Experience
Implement these designs for when a user edits her identity/profile.

**Mobile**
- [ ] https://zpl.io/agQ1mXM
- [ ] https://zpl.io/2ZrMWj1
- [ ] https://zpl.io/25JzRPr
- [ ] https://zpl.io/VQvyJpA
- [ ] https://zpl.io/2GGqP64
- [ ] https://zpl.io/2j5yqrr
- [ ] https://zpl.io/2pvgRzQ
- [ ] https://zpl.io/VkpQ5L5

**Desktop**
- [ ] https://zpl.io/b64dvMy
- [ ] https://zpl.io/bA7JEqB
- [ ] https://zpl.io/2GGJOQW
- [ ] https://zpl.io/2pvGBOj
- [ ] https://zpl.io/VDAJnzw
- [ ] https://zpl.io/29wxyZe

The key here is that the user be guided through the publishing step after each attestation or profile change.

### PR DESCRIPTION
**Pull request by shahthepro**
Should close #2415, #2421 and #2484

### Checklist
- [x] Get and replace images
- [x] User's profile page - #2415
- [x] Public profile - #2421 
- [x] ~~Publish button stays disabled - #2484~~ Publish button is no longer present in DApp3
- [ ] Push a test build and test on both mobile and desktop

### CODE REVIEW

**Comment by shahthepro** [comment]
@nick @sparrowDom I had to manually merge a few components during a merge conflict. I have deployed [a test build here](https://originprotocol.github.io/test-builds/shah). Can you check it out and let me know if I have broken some changes of yours?

I have tested the components for which conflict were there (MobileModalHeader, ListingCards, MobileModal). However I might have missed some styling.

**Comment by sparrowDom** [comment]
Screenshot 2019-06-19 13 11 52

Stuff in the test build looks nice, just noticed the images/caret-blue.svg missing. See back to Campaign link
Screenshot 2019-06-19 13 11 52

**Comment by shahthepro** [comment]
@sparrowDom That file exists in the branch though: https://github.com/OriginProtocol/origin/blob/shahthepro/dapp3-identity-profile-2415/dapps/marketplace/public/images/caret-blue.svg

File: dapps/marketplace/src/components/AttestationBadges.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  minCount = 10,
+  fillToNearest = 5,
+  onClick
+}) => {
+  let badges = providers || []
+
+  if (badges.length < minCount) {
+    badges = badges.concat(new Array(minCount - badges.length).fill(null))
+    // Show a minimum of `minCount` icons
+  } else if (badges.length > minCount && badges % fillToNearest !== 0) {
```
Comment (sparrowDom) [inline dapps/marketplace/src/components/AttestationBadges.js]: This should probably be: `else if (badges.length > minCount && badges.length % fillToNearest !== 0)` see the extra `length`

File: dapps/marketplace/src/components/Earning.js [hunk 2]
```diff
@@ -1,44 +1,79 @@
...
-  </div>
-)
+  )
+}
 
 export default Earnings
 
 require('react-styl')(`
   .earnings-progress
-    font-size: 18px
```
Comment (sparrowDom) [inline dapps/marketplace/src/components/Earning.js]: @nick wants to have all font sizes in px until we move to a better solution

File: dapps/marketplace/src/components/MobileModalHeader.js [hunk 3]
```diff
@@ -34,21 +35,31 @@ function MobileModalHeader({
     headerClassList = headerClassList.concat(className.split(' '))
   }
 
+  let closeButton
+
+  if (showCloseButton) {
```
Comment (sparrowDom) [inline dapps/marketplace/src/components/MobileModalHeader.js]: nitpick: 
```
if (showCloseButton || showBackButton) {
closeButton = (
      <a className={`modal-action-button ${showCloseButton ? 'close-button' : 'back-button'}`} onClick={onBack} />
    )
}
```

File: dapps/marketplace/src/components/Reviews.js [hunk 4]
```diff
@@ -98,14 +98,18 @@ export default class Reviews extends Component {
...
             <div className="reviews">
-              <h3>
-                {this.props.seller && (
-                  <fbt desc="reviews.headingSeller">Reviews of this seller</fbt>
-                )}
-                {!this.props.seller && (
-                  <fbt desc="reviews.headingUser">Reviews of this user</fbt>
-                )}
-              </h3>
+              {this.props.hideHeader ? null : (
```
Comment (sparrowDom) [inline dapps/marketplace/src/components/Reviews.js]: nitpick, another way to express this:
```
{!this.props.hideHeader && ( ....
```

File: dapps/marketplace/src/components/Reviews.js [hunk 5]
```diff
@@ -173,26 +182,41 @@ export default class Reviews extends Component {
...
             display: flex
             flex-direction: row
             align-items: baseline
             .name
-              font-size: 18px
+              font-family: Lato
+              font-size: 1.1rem
               font-weight: bold
-              color: var(--black)
+              font-style: normal
```
Comment (sparrowDom) [inline dapps/marketplace/src/components/Reviews.js]: From my experience in most cases these are not needed: 
```
              font-style: normal
              font-stretch: normal
              line-height: normal
              letter-spacing: normal
```

File: dapps/marketplace/src/components/TabView.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react'
+import get from 'lodash/get'
+
+// TBD: Notify Domen about this component
```
Comment (sparrowDom) [inline dapps/marketplace/src/components/TabView.js]: and you did :) thanks. I guess we can remove this now

File: dapps/marketplace/src/pages/nav/Nav.js [hunk 7]
```diff
@@ -38,6 +38,11 @@ const Nav = ({ location: { pathname }, isMobile, wallet, onGetStarted }) => {
   })
 
   if (isMobile) {
+    if (pathname.startsWith('/user')) {
```
[the remark this answers is not in the document]
Comment (shahthepro) [reply dapps/marketplace/src/pages/nav/Nav.js]: Can move it to `App.js`.

File: dapps/marketplace/src/components/AttestationBadges.js (hunk 1 above)
Comment (shahthepro) [reply dapps/marketplace/src/components/AttestationBadges.js]: Nice catch. Will fix it.

File: dapps/marketplace/src/components/Earning.js (hunk 2 above)
Comment (shahthepro) [reply dapps/marketplace/src/components/Earning.js]: Didn't know that. I have everything in rem right now. Should I convert them back to px? @nick

File: dapps/marketplace/src/components/MobileModalHeader.js (hunk 3 above)
Comment (shahthepro) [reply dapps/marketplace/src/components/MobileModalHeader.js]: Thanks. We can probably improve this thing further. Instead of having two boolean parameters (that are mutually exclusive), I wonder if we can have a single parameter that takes a one of string (constants/enums) values and render based on that.

File: dapps/marketplace/src/components/Reviews.js (hunk 5 above)
Comment (shahthepro) [reply dapps/marketplace/src/components/Reviews.js]: `line-height` probably affects a few places. I'll remove the other things if unnecessary.

File: dapps/marketplace/src/components/Earning.js (hunk 2 above)
Comment (nick) [reply dapps/marketplace/src/components/Earning.js]: It's not a big deal for now... but I want to replace all font sizes with vars so it's easier to be consistent throughout. In the meanwhile I've been using px for font sizes and rem for everything else

File: dapps/marketplace/src/components/Reviews.js [hunk 8]
```diff
@@ -173,26 +182,41 @@ export default class Reviews extends Component {
 require('react-styl')(`
   .reviews
     margin-bottom: 2rem
+    h3
+      font-family: Poppins
+      font-size: 1.5rem
+      font-weight: 500
+      font-style: normal
```
Comment (nick) [inline dapps/marketplace/src/components/Reviews.js]: It seems like having all these 'normal' declarations shouldn't be needed... isn't the default normal?

File: packages/graphql/src/mutations/identity/deployIdentity.js [hunk 9]
```diff
@@ -9,10 +9,14 @@ import costs from '../_gasCost.js'
 
 async function deployIdentity(
   _,
-  { from = contracts.defaultMobileAccount, profile = {}, attestations = [] }
```
Comment (nick) [inline packages/graphql/src/mutations/identity/deployIdentity.js]: why this change?

Comment (shahthepro) [reply packages/graphql/src/mutations/identity/deployIdentity.js]: Default values are returned only if `undefined` is passed ([Ref.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters)). For the `identity.attestations` and `identity.profile`, we get `null` from the GraphQL resolver.

File: dapps/marketplace/src/components/Reviews.js (hunk 8 above)
Comment (shahthepro) [reply dapps/marketplace/src/components/Reviews.js]: Actually, it inherits from parent/ancestor elements. If none of the elements have that property set, it should be `normal`. In some places, it is not necessary. I'll do a clean up of that.
